src/modules/currency-rate/currency-rate.repository.ts

```
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CurrencyRate } from './entity/currency-rate.entity';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class CurrencyRateRepository {
+  constructor(
+    @InjectRepository(CurrencyRate)
+    private readonly repository: Repository<CurrencyRate>,
+  ) {}
+
+  public findCurrencyRate(currency: string): Promise<CurrencyRate> {
+    return this.repository.findOneBy({ currency });
+  }
+
+  public upsertAll(currencyRates: CurrencyRate[]): void {
+    currencyRates.forEach(async (currencyRate) => {
+      await this.upsert(currencyRate);
+    });
+  }
+
+  private async upsert(currencyRate: CurrencyRate): Promise<void> {
+    currencyRate.updatedAt = new Date();
+
+    const existingPair = await this.repository.findOneBy({
+      currency: currencyRate.currency,
+    });
+
+    if (!existingPair) {
+      this.repository.insert(currencyRate);
+
+      return;
+    }
+
+    this.repository.update(
```

please `await` the update function, it also improves error logging, otherwise it might fail silently.